web-access-app/src/main/webapp/WEB-INF/jsp/fullRecord/aggregateRecord.jsp

```diff
@@ -145,7 +142,7 @@
                     <c:when test="${pdfViewerNeeded}">
                         <c:import url="fullRecord/pdfViewer.jsp" />
                     </c:when>
-                    <c:when test="${hasOriginalFileAccess && cdr:originalFileMimetypeMatches(briefObject, 'audio/(x-)?mpeg(-?3)?')}">
+                    <c:when test="${not empty dataFileUrl && audioPlayerNeeded}">
```

Comment:
just curious why audioPlayerNeeded needs to check for dataFileUrl first when pdfViewerNeeded doesn't?

Reply:
Hmm, I don't think it actually does need it. It should already be checked in hasPlayableAudioFile()

Reply:
Apparently shows the player if removed, but displays a null message in the player if no primary object set.

Reply:
For pdfs, it checks the primary object and then the first child if there is no primary object. You may need to do something similar for audio files. The two implementations would be pretty much identical aside from what mimetype to check for. Maybe they could use the same method internally and then hasPlayableAudioFile would pass the mimetype pattern in to the shared method? 

Alternatively, hasViewablePdf could be replaced by "getViewableDatastream" which would return the name of the datastream which is viewable. Internally it would do the first child solr query once and check against the mimetype patterns. Might be more scalable if we add more viewers, but on the other hand I think we're hoping that audio and video will become part of the UV display in the future.